ls1intum/Artemis
# `Development`: Ignore test courses for active user metrics

### Checklist
#### General


#### Server

### Motivation and Context
The metrics show information about active students in active courses and exams. Test courses never have any active students, they only have instructors/editors/tutors setting up and testing exercises.
Therefore, they should not be part of the calculated metrics.

### Description
Ignores test courses during metrics calculation.

Additionally optimises the database queries to ensure only relevant data is fetched.

### Steps for Testing (optional)

Prerequisites:
- 1 Admin

1. Enable metrics in your instance:
```yaml
management:
    metrics:
        export:
            prometheus:
                enabled: true
```
2. Make sure there are active courses and test courses in your Artemis instance.
3. The metrics should not contain any of the test courses.

### Review Progress






#### Performance Review
#### Code Review

### Test Coverage
Changed code covered and specific new tests added.
